src/test/less/parser.test.ts

```diff
@@ -16,11 +16,23 @@ suite('LESS - Parser', () => {
 	test('Variable', function () {
 		let parser = new LESSParser();
 		assertNode('@color', parser, parser._parseVariable.bind(parser));
+		assertNode('$color', parser, parser._parseVariable.bind(parser));
+		assertNode('$$color', parser, parser._parseVariable.bind(parser));
+		assertNode('@$color', parser, parser._parseVariable.bind(parser));
+		assertNode('$@color', parser, parser._parseVariable.bind(parser));
```

Comment:
This is a combination of two things.

1. Less supports variables and (like PHP) variable variables. Meaning, the name (identifier) of a variable being referenced can itself be variable.  So: `@@var` means "return the value of a variable with the id that `@var` resolves to. If `@var` is `foo` then `@foo` is looked up/returned.
2. In a 2.x release, support for property referencing (`$prop`) was added. Basically it just means that properties are (more or less) variables.

Well it's not explicitly documented, you can technically then combine these features of variable variables, where the property identifier being looked up is variable, or the variable id being looked up is variable based on a property name.

Why someone would ever do this, I don't know, it's just a normal side-effect of combining these features.

Reply:
This doesn't compile with less 3.9:

```lessc
@color: #fff;

@lookup: 'color';

.foo {
  color: @@lookup;
  background-color: $@lookup;
}
```

I guess we can add more less-specific language features in the future. Now as long as VS Code doesn't fail at parsing these less files, users can get by.

Reply:
@octref You're right that that usage may not compile, but this does:
```less
@dr: {
  prop: value;
}

.box {
  @lookup: prop;
  foo: @dr[$@lookup];
}
```
For lookup values, I think I did the VSCode parsing to use any of the variable reference form. So, yeah, it may not compile, but VSCode should still not throw an unrecoverable parsing error.

It's also possible the code you wrote will work at some point, just for consistency.